processor/attributekeyprocessor/attributekeyprocessor_test.go

```diff
@@ -0,0 +1,317 @@
+						Key:    "bar",
+						NewKey: "biz",
+					},
+					{
+						Key:    "foo",
+						NewKey: "bit",
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "duplicated_newKey",
+			args: args{
+				nextConsumer: nopProcessor,
+				replacements: []KeyReplacement{
+					{
+						Key:    "foo",
+						NewKey: "baz",
+					},
+					{
+						Key:    "bar",
+						NewKey: "biz",
+					},
+					{
+						Key:    "biz",
+						NewKey: "baz",
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "happy_path",
+			args: args{
+				nextConsumer: nopProcessor,
+				replacements: []KeyReplacement{
+					{
+						Key:    "foo",
+						NewKey: "baz",
+					},
+					{
+						Key:    "bar",
+						NewKey: "biz",
+					},
+				},
+			},
+			want: &attributekeyprocessor{
+				nextConsumer: nopProcessor,
+				replacements: []KeyReplacement{
+					{
+						Key:    "foo",
+						NewKey: "baz",
+					},
+					{
+						Key:    "bar",
+						NewKey: "biz",
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewTraceProcessor(tt.args.nextConsumer, tt.args.replacements...)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewTraceProcessor() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewTraceProcessor() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_attributekeyprocessor_ConsumeTraceData(t *testing.T) {
+	tests := []struct {
+		name string
+		args []KeyReplacement
+		td   data.TraceData
+		want []data.TraceData
+	}{
+		{
+			name: "keyMap_nil",
+			want: []data.TraceData{
+				{},
+			},
+		},
+		{
+			name: "span_nil",
+			args: []KeyReplacement{
+				{
+					Key:    "foo",
+					NewKey: "bar",
+				},
+			},
+			td: data.TraceData{
+				Spans: make([]*tracepb.Span, 1, 1),
+			},
+			want: []data.TraceData{
+				{
+					Spans: make([]*tracepb.Span, 1, 1),
+				},
+			},
+		},
+		{
+			name: "span_Attributes_nil",
+			args: []KeyReplacement{
+				{
+					Key:    "foo",
+					NewKey: "bar",
+				},
+			},
+			td: data.TraceData{
+				Spans: []*tracepb.Span{
+					{
+						Name: &tracepb.TruncatableString{Value: "test"},
+					},
+				},
+			},
+			want: []data.TraceData{
+				{
+					Spans: []*tracepb.Span{
+						{
+							Name: &tracepb.TruncatableString{Value: "test"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "span_Attributes_AttributeMap_nil",
+			args: []KeyReplacement{
+				{
+					Key:    "foo",
+					NewKey: "bar",
+				},
+			},
+			td: data.TraceData{
+				Spans: []*tracepb.Span{
+					{
+						Name:       &tracepb.TruncatableString{Value: "test"},
+						Attributes: &tracepb.Span_Attributes{},
+					},
+				},
+			},
+			want: []data.TraceData{
+				{
+					Spans: []*tracepb.Span{
+						{
+							Name:       &tracepb.TruncatableString{Value: "test"},
+							Attributes: &tracepb.Span_Attributes{},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "span_replace_key",
```

Comment:
Consider adding another test where both `Overwrite` and `KeepOriginal` for the same `KeyReplacement`.